### ISSUE TITLE
rewriteRelativeImportExtensions does not rewrite the extension for an import expressions unless there is an import declaration for the same module

### ISSUE THREAD

**Issue by bradzacher** [microsoft/TypeScript#60599]
### 🔎 Search Terms

rewriteRelativeImportExtensions, import expression

### 🕗 Version & Regression Information

- This changed between versions 5.7

### ⏯ Playground Link

https://www.typescriptlang.org/play/?rewriteRelativeImportExtensions=true#code/PTAEEsFsAcHsCcAuoDkA6YAzWs2IM4oDcAsAFBRxIAU6WOehAlOUA

### 💻 Code

Test One
```ts
import('./foo.ts')
```
Test Two
```ts
import './foo.ts';
import('./foo.ts')
```

### 🙁 Actual behavior

JS Output

Test One
```ts
import('./foo.ts') // ❌
```
Test Two
```ts
import './foo.js'; // ✅
import('./foo.js') // ✅
```

### 🙂 Expected behavior

JS Output

Test One
```ts
import('./foo.js') // ✅
```
Test Two
```ts
import './foo.js'; // ✅
import('./foo.js') // ✅
```

**Comment by jakebailey** [comment]
I don't think this has to do with the import being twice, but rather module detection? Tjis one rewrites paths too: https://www.typescriptlang.org/play/?rewriteRelativeImportExtensions=true#code/JYWwDg9gTgLgBAcgHQHoBmEICYkwM4IDcAsAFCiSwAUy6muBAlGUA

**Comment by jakebailey** [comment]
@andrewbranch

**Comment by jakebailey** [comment]
FWIW I'm not sure that this is incorrect behavior; this transformation was only intended on applying in modules, so if we don't think a file is a module, then it won't happen.

What condition are you running under where the file is not a module?

(Maybe we _should_ rewrite imports in `import` nodes in scripts, given they can appear there?)

**Comment by bradzacher** [comment]
If you're working in LTS node then the only way to access ESM in CJS is via `import` expressions.
But it's not a real usecase I had in mind -- I was just testing the feature and this surprised me because there was no mention of it being restricted on the announcement, and there aren't docs yet (https://www.typescriptlang.org/tsconfig/#rewriteRelativeImportExtensions :sad:).

**Comment by jakebailey** [comment]
To be clear a "module" to us means "not a global script"; CJS are modules in this scheme.
